### ISSUE TITLE
Private New RFD command referenced in public site

### ISSUE THREAD

**Issue by mattdailis** [oxidecomputer/rfd-site#64]
<img width="525" alt="Screenshot 2024-08-30 at 7 22 33 PM" src="https://github.com/user-attachments/assets/fe3d1831-c32d-4783-8277-6710af3cf8db">

`new.sh` is in the private rfd repo. I imagine it's relatively simple, and not proprietary, so it would be great if it could be made public!

Of course, it's convenient for it to be colocated with the `rfd` repo, so I don't think I'm proposing it be moved somewhere else. Perhaps an empty `rfd-template` repo for others to use as a starting point would be appropriate?

**Comment by david-crespo** [comment]
I think we just want to make this line conditional on some attribute of the user that indicates they’re internal.

https://github.com/oxidecomputer/rfd-site/blob/fc95e76510ff15ddd919ca651c6d67b3091b6e10/app/components/Header.tsx#L82

I’m sure we can share the script, but it very interesting: basically it creates a directory named 0123 containing a placeholder RFD asciidoc file (with the provided title stuck in it) in a branch also named 0123, and it pushes the branch.

**Comment by mattdailis** [comment]
Thanks! That's enough to get me started.

In case anyone else finds it useful to use as a starting point, here's my crude first cut based on David's description above: https://github.com/mattdailis/rfd-template

**Comment by augustuswm** [comment]
Agreed, we should limit that button to only users that have permission to create a new RFD. I have to get a new release of the api deployed but we have a permission variant that we should be able to check to verify that.

Related to the new rfd script, we don't have any plains to make that not work, we are finishing some work on api endpoints for creating a new RFD that does the same mechanical bits on behalf of the user. It is likely that we will update the create RFD button to show both options to the user.
